pkg/api/handler.go

```diff
@@ -56,7 +56,10 @@ type Handler interface {
 // Executor executes any aspect as described by config.Combined.
 type Executor interface {
 	// Execute performs actions described in combined config using an attribute bag.
-	Execute(cfg *config.Combined, attrs attribute.Bag) (*aspect.Output, error)
+	Execute(ctx context.Context, cfg *config.Combined, attrs attribute.Bag) (*aspect.Output, error)
```

Comment:
Why do we need both Execute and BulkExecute? Given the usage, I'd rather have a single method called Execute that takes a [].